pkg/config/validator.go

```diff
@@ -307,7 +312,48 @@ func validateSourceBindings(sl validator.StructLevel, sources map[string]Sources
 		}
 	}
 
-	validateBindPlugins(sl, enabledPluginsViaBindings)
+	validateBoundPlugins(sl, enabledPluginsViaBindings)
+	validatePluginRBAC(sl, sources, bindings)
+}
+
+func validatePluginRBAC[P providesPlugins](sl validator.StructLevel, hasPlugins map[string]P, bindings []string) {
+	// 1. identify duplicates
+	groups := make(map[string][]string)
+	for _, b := range bindings {
+		plugins := hasPlugins[b]
+		for pluginKey, plugin := range plugins.GetPlugins() {
+			if !plugin.Enabled {
+				continue
+			}
+			if g := groups[pluginKey]; len(g) == 0 {
+				groups[pluginKey] = []string{b}
+			} else {
+				groups[pluginKey] = append(groups[pluginKey], b)
+			}
+		}
+	}
+
+	// 2. compare RBAC of duplicates
+	for plugin, occurrences := range groups {
+		if len(occurrences) > 1 {
+			// take the head of occurrences
+			p1 := occurrences[0]
+			rbac1 := hasPlugins[p1].GetPlugins()[plugin].Context.RBAC
+			defaultNS1 := hasPlugins[p1].GetPlugins()[plugin].Context.Kubeconfig.DefaultNamespace
+			// compare the head with the tail
+			for i := 1; i < len(occurrences); i++ {
+				p2 := occurrences[i]
+				rbac2 := hasPlugins[p2].GetPlugins()[plugin].Context.RBAC
+				defaultNS2 := hasPlugins[p2].GetPlugins()[plugin].Context.Kubeconfig.DefaultNamespace
+				if !reflect.DeepEqual(rbac1, rbac2) {
+					sl.ReportError(bindings, p1, p1, invalidPluginRBACTag, p2)
+				}
+				if defaultNS1 != defaultNS2 {
+					sl.ReportError(bindings, p1, p1, invalidPluginDefaultNSTag, p2)
+				}
+			}
```

Comment:
1. Let's simplify the code:
- add a variable to avoid having `pluginConfigs[p2].GetPlugins()[plugin]` two times
- invert the if condition to reduce indentation
- avoid potential panics. Yes, they shouldn't happen in such case, but better to be on a safe side 👍 

Full code:
```go
        // 2. compare RBAC of duplicates
	for plugin, occurrences := range groups {
		if len(occurrences) < 2 {
			continue
		}

		// take the head of occurrences
		p1 := occurrences[0]
		p1Cfg, ok := pluginConfigs[p1].GetPlugins()[plugin]
		if !ok {
			continue
		}
		rbac1 := p1Cfg.Context.RBAC
		defaultNS1 := p1Cfg.Context.Kubeconfig.DefaultNamespace

		// compare the head with the tail
		for i := 1; i < len(occurrences); i++ {
			p2 := occurrences[i]
			p2Cfg, ok := pluginConfigs[p1].GetPlugins()[plugin]
			if !ok {
				continue
			}
			
			rbac2 := p2Cfg.Context.RBAC
			defaultNS2 := p2Cfg.Context.Kubeconfig.DefaultNamespace
			if !reflect.DeepEqual(rbac1, rbac2) {
				sl.ReportError(bindings, p1, p1, invalidPluginRBACTag, p2)
			}
			if defaultNS1 != defaultNS2 {
				sl.ReportError(bindings, p1, p1, invalidPluginDefaultNSTag, p2)
			}
		}

	}
```